boa/src/builtins/value/mod.rs

```diff
@@ -149,6 +150,95 @@ impl Value {
 
         Self::object(object)
     }
+
+    /// Convert from a JSON value to a JS value
+    pub fn from_json(json: JSONValue, interpreter: &mut Interpreter) -> Self {
+        match json {
+            JSONValue::Number(v) => {
+                Self::rational(v.as_f64().expect("Could not convert value to f64"))
+            }
```

Comment:
We have a small optimisation here that we could use. If the number is an integer, we store is a `Self::integer`, which helps with performance.

Reply:
Serde Json only stores numbers in 64 bits, but boa supports i32 and BigInt. How would you prefer to handle? Check to see if i64 can fit into i32; otherwise, use big int?

Reply:
Yes, I think that's the best option. Do a `try_from()` and if it fails, go for a rational.